Compilation fails on aarch64

Trying to compile latest version (ae9663f1887513e) on aarch64 with most recent GCC:

```
k_quants.c: In function ‘ggml_vec_dot_q2_K_q8_K’:
k_quants.c:1038:36: warning: missing braces around initializer [-Wmissing-braces]
 1038 |         const int16x8x2_t mins16 = {vreinterpretq_s16_u16(vmovl_u8(vget_low_u8(mins))), vreinterpretq_s16_u16(vmovl_u8(vget_high_u8(mins)))};
      |                                    ^
      |                                     {                                                                                                      }
k_quants.c: In function ‘ggml_vec_dot_q4_K_q8_K’:
k_quants.c:1522:5: note: use ‘-flax-vector-conversions’ to permit conversions between vectors with differing element types or numbers of subparts
 1522 |     const uint32x4_t mzero = vdupq_n_s32(0);
      |     ^~~~~
k_quants.c:1522:30: error: incompatible types when initializing type ‘uint32x4_t’ using type ‘int32x4_t’
 1522 |     const uint32x4_t mzero = vdupq_n_s32(0);
      |                              ^~~~~~~~~~~
k_quants.c:1567:54: error: incompatible type for argument 1 of ‘vdotq_s32’
 1567 |             const int32x4_t p1 = vdotq_s32(vdotq_s32(mzero, q4bytes.val[0], q8bytes.val[0]), q4bytes.val[1], q8bytes.val[1]);
      |                                                      ^~~~~
      |                                                      |
      |                                                      uint32x4_t
In file included from k_quants.c:14:
/usr/lib/gcc/aarch64-unknown-linux-gnu/12/include/arm_neon.h:29527:22: note: expected ‘int32x4_t’ but argument is of type ‘uint32x4_t’
29527 | vdotq_s32 (int32x4_t __r, int8x16_t __a, int8x16_t __b)
      |            ~~~~~~~~~~^~~
k_quants.c:1574:54: error: incompatible type for argument 1 of ‘vdotq_s32’
 1574 |             const int32x4_t p2 = vdotq_s32(vdotq_s32(mzero, q4bytes.val[0], q8bytes.val[0]), q4bytes.val[1], q8bytes.val[1]);
      |                                                      ^~~~~
      |                                                      |
      |                                                      uint32x4_t
/usr/lib/gcc/aarch64-unknown-linux-gnu/12/include/arm_neon.h:29527:22: note: expected ‘int32x4_t’ but argument is of type ‘uint32x4_t’
29527 | vdotq_s32 (int32x4_t __r, int8x16_t __a, int8x16_t __b)
      |            ~~~~~~~~~~^~~
k_quants.c: In function ‘ggml_vec_dot_q5_K_q8_K’:
k_quants.c:1807:52: error: incompatible type for argument 1 of ‘vdotq_s32’
 1807 |             sumi += vaddvq_s32(vdotq_s32(vdotq_s32(mzero, q5bytes.val[0], q8bytes.val[0]), q5bytes.val[1], q8bytes.val[1])) * *scales++;
      |                                                    ^~~~~
      |                                                    |
      |                                                    uint32x4_t
/usr/lib/gcc/aarch64-unknown-linux-gnu/12/include/arm_neon.h:29527:22: note: expected ‘int32x4_t’ but argument is of type ‘uint32x4_t’
29527 | vdotq_s32 (int32x4_t __r, int8x16_t __a, int8x16_t __b)
      |            ~~~~~~~~~~^~~
k_quants.c:1808:52: error: incompatible type for argument 1 of ‘vdotq_s32’
 1808 |             sumi += vaddvq_s32(vdotq_s32(vdotq_s32(mzero, q5bytes.val[2], q8bytes.val[2]), q5bytes.val[3], q8bytes.val[3])) * *scales++;
      |                                                    ^~~~~
      |                                                    |
      |                                                    uint32x4_t
/usr/lib/gcc/aarch64-unknown-linux-gnu/12/include/arm_neon.h:29527:22: note: expected ‘int32x4_t’ but argument is of type ‘uint32x4_t’
29527 | vdotq_s32 (int32x4_t __r, int8x16_t __a, int8x16_t __b)
      |            ~~~~~~~~~~^~~
k_quants.c: In function ‘ggml_vec_dot_q6_K_q8_K’:
k_quants.c:2010:38: warning: missing braces around initializer [-Wmissing-braces]
 2010 |         const int16x8x2_t q6scales = {vmovl_s8(vget_low_s8(scales)), vmovl_s8(vget_high_s8(scales))};
      |                                      ^
      |                                       {                                                            }
```
There are several errors and warnings.

lscpu (this is Oracle Cloud ARM instance):
```
Architecture:            aarch64
  CPU op-mode(s):        32-bit, 64-bit
  Byte Order:            Little Endian
CPU(s):                  4
  On-line CPU(s) list:   0-3
Vendor ID:               ARM
  Model name:            Neoverse-N1
    Model:               1
    Thread(s) per core:  1
    Core(s) per cluster: 4
    Socket(s):           -
    Cluster(s):          1
    Stepping:            r3p1
    BogoMIPS:            50.00
    Flags:               fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp ssbs
NUMA:                    
  NUMA node(s):          1
  NUMA node0 CPU(s):     0-3
Vulnerabilities:         
  Itlb multihit:         Not affected
  L1tf:                  Not affected
  Mds:                   Not affected
  Meltdown:              Not affected
  Mmio stale data:       Not affected
  Retbleed:              Not affected
  Spec store bypass:     Mitigation; Speculative Store Bypass disabled via prctl
  Spectre v1:            Mitigation; __user pointer sanitization
  Spectre v2:            Vulnerable: Unprivileged eBPF enabled
  Srbds:                 Not affected
  Tsx async abort:       Not affected
```

```
$ uname -a
Linux gnt 6.1.28-gentoo #3 SMP PREEMPT Fri Jun  9 14:27:51 UTC 2023 aarch64 GNU/Linux
$ python3 --version
Python 3.11.4
$ make --version
GNU Make 4.4.1
$ g++ --version
g++ (Gentoo 12.2.1_p20230428-r1 p2) 12.2.1 20230428
```

--- a/k_quants.c
+++ b/k_quants.c
@@ -1519,7 +1519,7 @@ void ggml_vec_dot_q4_K_q8_K(const int n, float * restrict s, const void * restri
 
     const uint8x16_t m4b = vdupq_n_u8(0xf);
 #ifdef __ARM_FEATURE_DOTPROD
-    const uint32x4_t mzero = vdupq_n_s32(0);
+    const int32x4_t mzero = vdupq_n_s32(0);
 #endif
 
     int8x16x2_t q4bytes;
@@ -1745,7 +1745,7 @@ void ggml_vec_dot_q5_K_q8_K(const int n, float * restrict s, const void * restri
 #ifdef __ARM_NEON
 
     const uint8x16_t m4b = vdupq_n_u8(0xf);
-    const uint32x4_t mzero = vdupq_n_u32(0);
+    const int32x4_t mzero = vdupq_n_s32(0);
     const uint8x16_t mone = vdupq_n_u8(1);
     const uint8x16_t mtwo = vdupq_n_u8(2);
 
@@ -2242,5 +2242,3 @@ void ggml_vec_dot_q6_K_q8_K(const int n, float * restrict s, const void * restri
     *s = sumf;
 #endif
 }
-
-

